### ISSUE TITLE
Emit a diagnostic on `NamedTuple` classes decorated with `@dataclass`

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#2515]
If a user does this, it indicates that they're confused, because attempting to create an instance of a class like this always fails at runtime:

```py
from dataclasses import dataclass
from typing import NamedTuple

@dataclass
class Foo(NamedTuple):
    x: int
    y: str

# Both of these calls fail with `AttributeError: can't set attribute`
Foo(42, "y")
Foo(x=42, y="y")
```

We don't model the semantics here correctly right now (we don't synthesize any of the `NamedTuple` generated methods on this class at all -- we think, for instance, that this class inherits its `__new__` method from `tuple`). But we don't need to model the semantics correctly if we detect that this is an error (which it is) and emit an appropriate diagnostic.

We added a TODO for this in the code in https://github.com/astral-sh/ruff/commit/3e0299488e72fa80f59473565cf724041d0f3f3c, but I don't think we added any tests for it.
